typing/tast_iterator.ml

```diff
@@ -68,6 +68,20 @@ type iterator =
 let iter_snd f (_, y) = f y
 let iter_loc sub {loc; _} = sub.location sub loc
 
+let rec iter_loc_lid sub lid =
+  let open Longident in
+  match lid with
+  | Lident _ -> ()
+  | Ldot (lid, id) ->
+      iter_loc sub lid; iter_loc_lid sub lid.txt; iter_loc sub id
+  | Lapply (lid, lid') ->
+      iter_loc sub lid; iter_loc_lid sub lid.txt;
+      iter_loc sub lid'; iter_loc_lid sub lid'.txt
+
+let iter_loc_lid sub {loc; txt} =
+  iter_loc_lid sub txt;
+  iter_loc sub {loc; txt}
```

Comment:
Same remark as on the parsetree iterator: iterating first on the full location before iterating on the sublocation feels slightly more consistent to me.